rskj-core/src/test/java/co/rsk/peg/InMemoryStorageSimulator.java

```diff
@@ -0,0 +1,41 @@
+package co.rsk.peg;
+
+import co.rsk.peg.storage.StorageAccessException;
+import co.rsk.peg.storage.StorageAccessor;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.ethereum.vm.DataWord;
+
+/**
+ * This is an In-Memory Storage Simulator to be used in Unit Tests to reduce the use of mocks, thus
+ * improving the Unit Tests reliability.
+ */
+public class InMemoryStorageSimulator implements StorageAccessor {
```

Comment:
The way I see it is that the Get and Put will handle the same type for each instance of the class, if that is not the case I would be careful with casting errors that could occur. 
I would recommend parametrizing the `InMemoryStorageSimulator` class by the type `T` so that the class is more flexible and type-safe. Wdyt?

Reply:
I agree. The class should be parameterized: `InMemoryStorageSimulator<T>`.

Reply:
But the class willl not always handle the same data type. It can be used to store Coin, Federation, UTXO, etc